README.md

```diff
@@ -1,3 +1,203 @@
+```
+brew install git-crypt
+cd ~/sandbox/data-science-sandbox-infrastucture
+git-crypt unlock
+```
+
+Now install kubectl:
+```
+brew install kubectl  # or on ubuntu we do: sudo snap install kubectl --classic
+```
+For kubectl to talk to k8s we to use kops to get the secret from S3 and write out `~/.kube/config`.
+First Dan gives you access to the S3 bucket. Then:
+```
+brew install kops
+export KOPS_STATE_STORE=s3://kops.data-science.org.uk
+export KOPS_NAME=sandbox.data-science.org.uk
+kops export kubecfg --name sandbox.data-science.org.uk
+# creates ~/.kube/config used by kubectl
+kubectl config set-cluster default-cluster --server=https://dashboard.services.sandbox.data-science.org.uk
+```
+
+Now this should list the sandboxes:
+```
+kubectl get namespaces
+```
+
+Kubectl gives fundamental access to k8s. Now we install 'helm' which is a higher level interface - it deals in 'charts' which are packages, making it easy to install and configure bunches of k8s resources, such as rstudio.
+
+We need old version of helm, to match the version on our k8s cluster.
+```
+DESIRED_VERSION=v2.3.0 bash -c 'curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get | bash'
+# helm installed into /usr/local/bin/helm
+helm init
+```
+
+We need a helm chart that MOJ provides - it is more up to date than the one in our fork:
+```
+helm repo add mojanalytics https://ministryofjustice.github.io/analytics-platform-helm-charts/charts/
+```
+
+Playing with sandboxes using helm:
+```
+export KUBECONFIG=~/.kube/config
+export USERNAME=andreagrandi
+export EMAIL=andrea.grandi@<domain>.gov.uk
+export FULLNAME=Andrea Grandi
+export PLATFORM_ENV=sandbox
+export HELM=helm
+cd ~/sandbox/data-science-sandbox-infrastucture
+
+e.g.
+$HELM delete init-user-${USERNAME} --purge
+$HELM delete ${USERNAME}-rstudio --purge
+```
+
+### This repo
+
+Clone this repo:
+```
+cd ~/sandbox
+git clone https://github.com/ukgovdatascience/sandbox-deploy.git
+```
+
+Create a virtualenv:
+```
+mkvirtualenv --python=/usr/local/bin/python3 sandbox-deploy
+```
+Edit ~/.virtualenvs/sandbox-deploy/bin/postactivate so it contains:
+```
+cd ~/sandbox/sandbox-deploy
+export FLASK_APP=deploy/deploy.py
+export FLASK_DEBUG=true
+export SANDBOX_DEPLOY_USERNAME=
+```
+Install the dependencies etc:
+```
+workon sandbox-deploy
+cd ..
+git clone
+pip install -r requirements.txt
+```
+
+### Use this repo
+
+Always activate the environment:
+```
+workon sandbox-deploy
+```
+
+The deploy script can be run on its own:
+```
+USERNAME=davidread deploy/deploy.sh
+```
+
+python deploy.py  #etc
+
+# examples
+python3 deploy/commands.py deploy
+python3 deploy/commands.py deploy --username=davidread --email='david.read@<domain>.gov.uk' --fullname='D Read'
+
+# run deploy app
+workon sandbox-deploy
+
+flask run --port 7000
+
+
+## Deployment to EC2
+
+```
+# join GDS vpn
```

Comment:
(not needed if we run this from the office, we should probably specify to avoid a new user spend any time configuring the VPN if just in the office)

Reply:
good point - I'll change it

Reply:
good point - I'll change this